### PR TITLE
fix(geo): Add lib overrides for mapbox-gl

### LINKED ISSUE #1656: CRA 4 build failure with the new map component
### Before creating a new issue, please confirm:

- [X] I have [searched for duplicate or closed issues](https://github.com/aws-amplify/amplify-ui/issues?q=is%3Aissue+) and [discussions](https://github.com/aws-amplify/amplify-ui/discussions).
- [X] I have read the guide for [submitting bug reports](https://github.com/aws-amplify/amplify-ui/blob/main/CONTRIBUTING.md#bug-reports).

### On which framework/platform are you having an issue?

React

### Which UI component?

Other

### How is your app built?

CRA 4

### Please describe your bug.

Fresh application fails to start with a compile error when using CRA 4 showing the compile error 

```
./node_modules/react-map-gl/dist/esm/components/use-control.js 12:46
Module parse failed: Unexpected token (12:46)
File was processed with these loaders:
 * ./node_modules/babel-loader/lib/index.js
You may need an additional loader to handle the result of these loaders.
| 
|     if (!map.hasControl(ctrl)) {
>       map.addControl(ctrl, (opts || onRemove)?.position);
|     }
|
```

This issue is resolved when using the current version of CRA 5.

### What's the expected behaviour?

No compile errors on run.

### Help us reproduce the bug!

using CRA 4 create a react application
`npm install aws-amplify @aws-amplify/ui-react`
import any component from the ui-react package into your application
`npm run start` to see the compile error

### Code Snippet

```typescript
// Put your code below this line.

```

### Additional information and screenshots

package dependencies
```
"dependencies": {
    "@aws-amplify/ui-react": "^2.15.1",
    "@testing-library/jest-dom": "^5.16.4",
    "@testing-library/react": "^12.1.4",
    "@testing-library/user-event": "^13.5.0",
    "aws-amplify": "^4.3.19",
    "react": "^18.0.0",
    "react-dom": "^18.0.0",
    "react-scripts": "4.0.3",
    "web-vitals": "^2.1.4"
  },
```

_No response_

### LINKED ISSUE #1736: CRA 3.4.1 and 4.0.3 prod build fails on `mapbox-gl` transitive dependency
### Before creating a new issue, please confirm:

- [X] I have [searched for duplicate or closed issues](https://github.com/aws-amplify/amplify-ui/issues?q=is%3Aissue+) and [discussions](https://github.com/aws-amplify/amplify-ui/discussions).
- [X] I have read the guide for [submitting bug reports](https://github.com/aws-amplify/amplify-ui/blob/main/CONTRIBUTING.md#bug-reports).

### On which framework/platform are you having an issue?

React

### Which UI component?

Other

### How is your app built?

CRA or 3.4.1 or 4.0.3

### Please describe your bug.

```
Creating an optimized production build...
Failed to compile.
./node_modules/react-map-gl/dist/esm/components/map.js
Cannot find module: 'mapbox-gl'. Make sure this package is installed.
You can install this package by running: npm install mapbox-gl.
```

### What's the expected behaviour?

Should build without issues

### Help us reproduce the bug!

Do a prod build with CRA 3 or 4 and `@aws-amplify/ui-react` package.

### Code Snippet

```typescript
// Put your code below this line.

```

### Additional information and screenshots

_No response_

### PR DESCRIPTION
**Pull request by slaymance**
#### Description of changes
This PR adds lib overrides for the `mapbox-gl` dependency. `lib/mapbox-gl` will resolve `mapbox-gl` to an empty file so that the package can be resolved for users using bundlers like Webpack. This will resolve #1736.

#### Issue #, if available
#1736 

#### Description of how you validated changes
Tested this change manually in a CRA and Next.js application. To reproduce manual testing:
1. Pull this branch down and run `yarn setup`
2. In the `packages/react/` directory, run `npm pack --pack-destination ~/Desktop`
3. In an Amplify UI sample application that uses Webpack for bundling (Next, CRA, etc.), install the tarball generated in the step above as a depenency: `npm install ~/Desktop/aws-amplify-ui-react-2.19.0.tgz` (note you must install it with `npm` since we used that to pack the tarball)
5. Build the sample application and ensure it compiles correctly. Inspect `node_modules/mapbox-gl` and ensure it matches `lib/mapbox-gl` in this PR. You may need to delete your `node_modules` and lock files and reinstall dependencies.

#### Checklist

- [x] PR description included
- [x] `yarn test` passes
- [x] Tests are updated - N/A
- [x] No side effects or [`sideEffects`](https://github.com/aws-amplify/amplify-ui/blob/main/packages/react/CONTRIBUTING.md#code-standards) field updated
- [x] Relevant documentation is changed or added (and PR referenced)

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

File: packages/react/lib/react-map-gl/package.json [hunk 1]
```diff
@@ -0,0 +1,8 @@
+{
+  "name": "react-map-gl",
+  "version": "0.0.0",
+  "main": "node_modules/react-map-gl/dist/es5/index.js",
```
Comment (reesscot) [inline packages/react/lib/react-map-gl/package.json:4]: Won't this fail to resolve once it's no longer in the monorepo?

File: packages/react/package.json [hunk 2]
```diff
@@ -68,7 +69,7 @@
     "classnames": "2.3.1",
     "deepmerge": "4.2.2",
     "lodash": "4.17.21",
-    "mapbox-gl": "npm:empty-npm-package@1.0.0",
+    "mapbox-gl": "file:lib/mapbox-gl",
```
Comment (zchenwei) [inline packages/react/package.json:72]: Does that mean `file` alias is more widely accepted than `npm` alias?

**Review by zchenwei — APPROVED** [review]
LGTM! With a question.

File: packages/react/package.json (hunk 2 above)
Comment (slaymance) [reply packages/react/package.json:72]: We've had some problems when using hosted apps with the `npm` prefix (like npm pretty much won't resolve the package as aliased before). I've verified that the `file:` prefix works in a hosted Amplify Console application.

**Comment by slaymance** [comment]
Cosed in favor of #2061

**Pull request closed without merging**